test/league_test.js

```diff
@@ -14,5 +14,290 @@ describe('league', function () {
       expect(players).to.have.lengthOf(1);
       expect(players[0]).to.have.members(['Bob']);
     });
+
+    it('players are added from top row downwards', function () {
+      const league = gameState.createLeague();
+      league.addPlayer('Player1');
+      const players1 = league.getPlayers();
+      league.addPlayer('Player2');
+      league.addPlayer('Player3');
+      const players2 = league.getPlayers();
+      league.addPlayer('Player4');
+      league.addPlayer('Player5');
+      league.addPlayer('Player6');
+      const players3 = league.getPlayers();
+
+      expect(players1).to.have.lengthOf(1);
+      expect(players1[0][0]).to.equal('Player1');
+      expect(players2).to.have.lengthOf(2);
+      expect(players2[0][0]).to.equal('Player1');
+      expect(players2[1][0]).to.equal('Player2');
+      expect(players2[1][1]).to.equal('Player3');
+      expect(players3).to.have.lengthOf(3);
+      expect(players3[0][0]).to.equal('Player1');
+      expect(players3[1][0]).to.equal('Player2');
+      expect(players3[1][1]).to.equal('Player3');
+      expect(players3[2][0]).to.equal('Player4');
+      expect(players3[2][1]).to.equal('Player5');
+      expect(players3[2][2]).to.equal('Player6');
+    });
+
+    it('doesn\'t add players with repeated names', function () {
+      const league = gameState.createLeague();
+      const repeatedName = 'Alice';
+
+      const addingNameTwice = function (){
+        league.addPlayer(repeatedName);
+        league.addPlayer(repeatedName);
+      };
+
+      expect(addingNameTwice).to.throw(`Cannot add player '${repeatedName}' because they are already in the game`);
+    });
+
+    it('doesn\'t add players with names containing spaces', function () {
+      //Given
+      const league = gameState.createLeague();
+      const invalidName = 'Bob Bobson';
+
+      //When
+      const addingInvalidName = function (){
+        league.addPlayer(invalidName);
+      };
+
+      //Then
+      expect(addingInvalidName).to.throw(`Player name ${invalidName} contains invalid characters`);
+    });
+
+    it('doesn\'t add players when no name is entered', function () {
+      //Given
+      const league = gameState.createLeague();
+      const noName = '';
+
+      //When
+      const addingNoName = function (){
+        league.addPlayer(noName);
+      };
+
+      //Then
+      expect(addingNoName).to.throw(`Player name ${noName} contains invalid characters`);
```

Comment:
This test accurately captures the behaviour of the current system, but also highlights a shortcoming. No invalid characters have been entered! The thrown error message should be improved to handle this specific case.